forem/forem
# Remove language settings

## What type of PR is this? (check all applicable)


## Description

While doing research for an RFC, I found some unused/unfinished code and we decided to clean that up. 

This PR does the following:

1. Remove `language_settings` and all related code.  Even though this was marked as "beta", it seems a bit too unfinished to provide any real benefit. The list of 7 languages seems very arbitrary: for example, we included only 4 out of the 10 most spoken languages, with Japanese being the only non-European one. If we revisit this future we probably also want to design the data model a bit better and have a separate table for this, not a JSONB column on `users`.
2. Usage of the `language` column on `Article`. It seems it was primary based on scoring an article's language against the user's preferred language(s), so with this feature gone it doesn't make sense to keep around. 

This whole topic can and probably should be revisited in the future, but with a proper RFC.

There will also be follow-up PRs to ignore and remove the two relevant columns.

Questions:

* [ ] @mstruve I removed the relevant attributes from ES indexing, but IIRC we can't remove them from the mapping, is that correct? 

## Related Tickets & Documents

https://github.com/forem/forem/issues/12356

## QA Instructions, Screenshots, Recordings

While this is a big change, it should primarily be covered by existing specs. 

For manual QA you can visit `settings/customization` and browse some articles and the feed to ensure everything is still working.

### UI accessibility concerns?

None

## Added tests?

- [X] Yes, updated/removed existing ones

## Added to documentation?

- [X] Update [Developer Docs](https://docs.forem.com)
